src/renderers/dom/shared/escapeTextContentForBrowser.js

```diff
@@ -32,7 +108,16 @@ function escaper(match) {
  * @return {string} An escaped string.
  */
 function escapeTextContentForBrowser(text) {
-  return ('' + text).replace(ESCAPE_REGEX, escaper);
+  switch (typeof text) {
```

Comment:
Actually, can we inline `if (typeof text === 'boolean' || typeof text === 'number')`? V8, at least, inlines `typeof foo === <string>` in the bytecode.